package/MDAnalysis/core/universe.py

```diff
@@ -1459,13 +1488,98 @@ def from_smiles(cls, smiles, sanitize=True, addHs=True,
                 "hydrogens with `addHs=True`")
 
             numConfs = rdkit_kwargs.pop("numConfs", numConfs)
-            if not (type(numConfs) is int and numConfs > 0):
+            if not (isinstance(numConfs, int) and numConfs > 0):
                 raise SyntaxError("numConfs must be a non-zero positive "
-                "integer instead of {0}".format(numConfs))
+                                  "integer instead of {0}".format(numConfs))
             AllChem.EmbedMultipleConfs(mol, numConfs, **rdkit_kwargs)
 
         return cls(mol, **kwargs)
 
+    def guess_TopologyAttributes(
+            self, context=None, to_guess=(), force_guess=(), **kwargs):
+        """
+        Guess and add attributes through a specific context-aware guesser.
+         .. _guess_TopologyAttributes:
+
+        Parameters
+        ----------
+        context: string or Guesser class
+            For calling a matching guesser class for this specific context
+        to_guess: list, (optional)
+            Attributes to be guessed (these attributes will be either guessed
+            if they don't exist in the universe or partially guessed by only filling
+            its empty values if universe has the attribute)
+        force_guess: list, (optional)
+            Attributes to be forced guessed (these attributes will be either
+            guessed if they don't exist in the universe or their values will be
+            completely overwritten by guessed values if the universe has the
+            attribute)
+        **kwargs: extra arguments to be passed to the guesser class
+
+        Examples
+        --------
+        guess ``masses`` and ``types`` attribute::
+
+            u.guess_TopologyAttributes(context='default', to_guess=['masses', 'types'])
+
+        .. versionadded:: 2.7.0
+
+        """
+        if not context:
+            context = self._context
+
+        guesser = get_guesser(context, self.universe, **kwargs)
+        self._context = guesser
+
+        total_guess = list(force_guess) + list(to_guess)
+
+        # removing duplicate from the guess list with keeping attributes order
+        # as it is more convientent to guess attributes
+        # in the same order that the user provided
+        total_guess = list(dict.fromkeys(total_guess))
+
+        objects = ['bonds', 'angles', 'dihedrals', 'impropers']
+
+        if(total_guess):
+            if (self._topology.n_atoms > 0):
+
+                if guesser.are_guessable(total_guess):
+
+                    for attr in to_guess:
+                        if any(attr == a for a in self._topology_atrrs):
```

Comment:
```suggestion
                if guesser.are_guessable(total_guess):
                    topology_attrs = [att.attrname for att in self._topology.read_attributes]

                    for attr in to_guess:
                        if any(attr == a for a in topology_attrs):
```

I believe here is the only place you use `self._topology_atrrs` -- could you instead create the list here instead of assigning a property (an example suggested here)

Reply:
Another way to check this is also using sets -- for example, instead of looping over each `attr in to_guess`, you could turn that into a set with `set(to_guess)` and check if there's an intersection with `set(topology_attr)`